### PR TITLE
feat: added more indent queries

### PR DESCRIPTION
**Pull request by elianiva**
I've added indent queries for some languages but I might have not covered some cases, would be awesome if anyone would try it and give some advice :)

List of languages that I've added
- HTML
- JSON
- YAML
- TOML
- Javascript
- Typescript
- JSX
- TSX
- Go
- Ruby
- GraphQL
- Rust
- Teal
- C
- C++

Known issues: 
- Last `/>` on self closing HTML tag doesn't get outdented properly **(Fixed after #837)**
- Last `</>` on jsx fragment doesn't get outdented properly
- Last `}` on Rust `else_clause` doesn't get outdented properly **(Fixed)**
- Last `end` on Teal sometimes doesn't get outdented properly. e.g if you use anonymous function for an argument.
- Some block doesn't get indented properly when you press `o` on the first line of the block to add a new line. Hopefully these illustrations make it clearer.
```javascript
return(| <- if you press `o` here
| <- your cursor will go here
  | <- instead of here, notice the indentation 
)
```
another case is the Rust's else clause
```rust
} else {| <- if you press `o` here
| <- it will go here, no indent
  println!("something");| <- if you press `o` here
  | <- it will go here, the indentation is correct
}
```

note: this branch will only receives force push commits

### CODE REVIEW

**Review by vigoux — APPROVED** [review]
Terrific work again :wink:
I'll let @steelsojka review javascript ones.
But for `rust` is seems okay, appart for the `else` dedenting, maybe we can implement a `@dedent` capture to force this behavior somehow.

File: queries/javascript/indents.scm
Comment (yzia2000) [inline queries/javascript/indents.scm:1]: jsx indents are not being imported in react js files. Works if you add indent and branch queries from jsx to javascript index.scm instead of inheritting. Might be due to some overriding

File: queries/javascript/indents.scm
Comment (yzia2000) [inline queries/javascript/indents.scm:24]: Indentation issue with else block:
```
if () {
} else {
    }
```
can be fixed by appending @ branch with statement_block

File: queries/tsx/indents.scm
Comment (yzia2000) [inline queries/tsx/indents.scm:1]: jsx components are working perfectly here

File: queries/typescript/indents.scm
Comment (yzia2000) [inline queries/typescript/indents.scm:1]: Indentation issue with else block:
```
if () {
} else {
    }
```

File: queries/go/indents.scm
Comment (yzia2000) [inline queries/go/indents.scm:16]: Indentation issue with else block:
```
if () {
} else {
    }
```
can be fixed by appending @ branch with block

**Review by yzia2000 — CHANGES_REQUESTED** [review]
jsx indents dont seem to be working when inheritted in js files. Also adding (statement_block) to javascript/indents and (block) to go/indents in @ branch fixes the indentation issue:
```
if () {
} else {
    }
```
Overall great work. Much needed for main branch.

**Comment by yzia2000** [comment]
Hi I have found some issues after reviewing this branch with go and javascript. Thanks for the work. This pr is much needed for development.

File: queries/javascript/indents.scm
Comment (elianiva) [reply queries/javascript/indents.scm:1]: hmm, it works on my end though. Let me double check it with a minimal init.vim

Comment (yzia2000) [reply queries/javascript/indents.scm:1]: Sure

File: queries/javascript/indents.scm
Comment (elianiva) [reply queries/javascript/indents.scm:24]: that does seem to work, but I found another issue with it. It gives the `else` block a double indentation like this.
```javascript
if (2 > 1) {
  console.log("A")
} else {
    console.log("c")
}
```

**Comment by yzia2000** [comment]
> Terrific work again 
> I'll let @steelsojka review javascript ones.
> But for `rust` is seems okay, appart for the `else` dedenting, maybe we can implement a `@dedent` capture to force this behavior somehow.

Rust dedenting seems to be working for me. I think the issue arises because alternative in js and go files are blocks or statement_blocks instead of else_blocks. Seems to work if we append @ branch with (block) and (statement_block) in the respective languages.

File: queries/javascript/indents.scm
Comment (yzia2000) [reply queries/javascript/indents.scm:24]: Really. It seems to work on my end. Let me debug again

Comment (yzia2000) [reply queries/javascript/indents.scm:24]:
```
        if (!this.state.toggled) {
            this.props.toggleOn();
        } else {
            this.props.toggleOff();
        }
```
I am getting this kind of indentation. Let me use minimal nvim

File: queries/javascript/indents.scm
Comment (elianiva) [reply queries/javascript/indents.scm:1]: it does seem to work using this minimal config https://paste.sh/OIGOYy-6#0Djftco4hcAZ2dgfN2BA0kTI, don't forget to set the `indentexpr` to `nvim_treesitter#indent()`. Not sure why it has to be done it manually.

Also, the last `</>` for jsx fragment seems to be broken, it doesn't get outdented properly. I thought by adding `"</>"` to `@branch` would fix it but it didn't.

File: queries/javascript/indents.scm
Comment (elianiva) [reply queries/javascript/indents.scm:25]: yep, tried your query and it still doesn't work unfortunately.

File: queries/javascript/indents.scm
Comment (yzia2000) [reply queries/javascript/indents.scm:1]: setting indentexpr made it work haha

File: queries/javascript/indents.scm
Comment (yzia2000) [reply queries/javascript/indents.scm:25]: actually I realized that it works without setting indentxpr and doesnt when indentxpr is explicitly set

Comment (elianiva) [reply queries/javascript/indents.scm:25]: I don't know why it doesn't set it automatically when `indent.enable` is set to true. I believe it was set automatically in the past but then it stops doing that at some point (or I just misremember it, idk)

Comment (elianiva) [reply queries/javascript/indents.scm:25]: ahhhh, so that's the issue. Fixed it, thanks!

**Comment by elianiva** [comment]
Question: does Teal has the same indentation as Lua? As far as I know, Teal is just Lua but strongly typed (and a simpler one than  JS compared TS). If it's the same then I can just copy the existing Lua indents for Teal.

**Comment by elianiva** [comment]
Huh, strange. Teal indentation doesn't work at all. I'll push them and see if anyone also has this issue.

File: queries/go/indents.scm
Comment (yzia2000) [inline queries/go/indents.scm:13]: I think this should be put in indent

File: queries/javascript/indents.scm
Comment (yzia2000) [inline queries/javascript/indents.scm:18]: Not needed anymore I think

**Review by yzia2000 — CHANGES_REQUESTED** [review]
I think these two can be reverted now that if statement has been removed.

**Comment by elianiva** [comment]
@vigoux I tried fixing the outdent for `}` on else clause and I think I've fixed it. Does it work for you?

**Comment by vigoux** [comment]
We'll wait after #837 is merged to merge this, as we shoudl first fix the core mechanism before trying to find work-arounds in the queries.

**Comment by vigoux** [comment]
Could you try rebasing ? And is this PR ready, I think we can go incrementally afterwards.

**Comment by elianiva** [comment]
Yup, I think this PR is ready. I've been using it daily and have no issue except the ones I've mentioned that we can improve overtime.

**Comment by vigoux** [comment]
We merged #837 so maybe most of these are actually fixed now.

**Comment by vigoux** [comment]
Could you remove the WIP tags, I'll merge that once it's done/

**Comment by elianiva** [comment]
done

**Pull request merged**
